tensilelite/Tensile/KernelWriter.py

```diff
@@ -1274,8 +1274,13 @@ def hasAnyDependency(lr: DSLoadInstruction, insts: List[Instruction]):
         ####
         # scheduled pointer
         ####
+        isDTL = kernel["DirectToLdsA"] or kernel["DirectToLdsB"]
         if mfmaIndex == self.states.lwEndMfmaIndex:
           iterCode.add(pointerLWCode)
+          # Wait for previous set of global reads to complete
+          if isDTL and not NLLlast:
+            vmcntVal = 1 if kernel["PrefetchGlobalRead"] == 2 else 0
+            iterCode.add(self._wait(kernel, tPA, tPB, vmcntVal, -1, -1, "wait for previous set of global reads"))
```

Comment:
Is this generated later than the previous location?
DTL case, global read wait should come with syncThread.
If you would like to move wait for DTL, you should move both wait and syncThread.

Reply:
It might be better to generate this wait code at syncPlrMfmaIndex.
We can skip syncThread at the previous location in DTL case.

Reply:
Sorry. syncThread code is stored in syncCode.
This is OK.

Reply:
Probably, we need this wait only for PGR2. PGR1 case, I think wait is generated at syncThread. Please confirm.

Reply:
> Is this generated later than the previous location?

Yeah, previously the wait was generated at the loop iteration boundary. This caused issues when the local writes crossed that boundary.

> Probably, we need this wait only for PGR2. PGR1 case

We still need this wait for PGR1 + DTL. I initially tried the wait with only PGR2+DTL but there were failures with PGR1 cases in the current dtl tests.

> I think wait is generated at syncThread. 

Okay, `syncThread` does generate a wait for PGR1 but there are edge cases where it does not ([example 1wave/WG](https://github.com/ROCm/hipBLASLt/blob/develop/tensilelite/Tensile/KernelWriterModules.py#L131)) which caused the failures I mentioned above.

Reply:
Another thing is about redundant wait code for PGR1 and num Wave > 1.
Would you please add some logic to avoid generating duplicated wait in that case?
Either removing wait in Sync for DTL or using skipForceWaitcnt0 for DTL.

Reply:
Thanks for the suggestions! :) 
Fixed.